repository/src/main/java/org/alfresco/repo/event2/DirectEventSender.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * #%L
+ * Alfresco Repository
+ * %%
+ * Copyright (C) 2005 - 2023 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.repo.event2;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import org.alfresco.error.AlfrescoRuntimeException;
+import org.alfresco.repo.event.v1.model.RepoEvent;
+import org.alfresco.util.PropertyCheck;
+import org.gytheio.messaging.MessagingException;
+import org.springframework.beans.factory.InitializingBean;
+
+/**
+ * Sender allows to send asynchronous events directly to topic.
```

Comment:
Events are by nature asynchronous. Also, this class doesn't care or know if the message destination is a topic or queue. I'd suggest to change the text to something like:
```Sends a message to a destination in the current thread```

Reply:
Fixed. Thanks